consensus/misc/eip1559.go

```diff
@@ -102,13 +102,75 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
 }
 
 // CalcBaseFee calculates the basefee of the header.
-func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseFee *big.Int) *big.Int {
+func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseFee *big.Int, currentHeaderTime uint64) *big.Int {
 	if config.Clique != nil && config.Clique.ShadowForkHeight != 0 && parent.Number.Uint64() >= config.Clique.ShadowForkHeight {
 		return big.NewInt(10000000) // 0.01 Gwei
 	}
 
-	scalar, overhead := ReadL2BaseFeeCoefficients()
-	return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	if parent == nil || parent.Number == nil || !config.IsFeynman(currentHeaderTime) {
+		scalar, overhead := ReadL2BaseFeeCoefficients()
+		return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	}
+	return calcBaseFeeFeynman(config, parent)
+}
+
+// calcBaseFeeFeynman calculates the basefee of the header for Feynman fork.
+func calcBaseFeeFeynman(config *params.ChainConfig, parent *types.Header) *big.Int {
+	baseFeeEIP1559 := calcBaseFeeEIP1559(config, parent)
+	baseFee := new(big.Int).Set(baseFeeEIP1559)
+	baseFee.Add(baseFee, config.ProvingBaseFee)
```

Comment:
What the final decision to use a hardcoded `config.ProvingBaseFee`? I thought we'd keep using `overhead` from `ReadL2BaseFeeCoefficients`.

Reply:
oh OK. discussed here, I thought it would be kept as it is: https://github.com/scroll-tech/go-ethereum/pull/1199#discussion_r2156569913.